### ISSUE TITLE
Eclipse IDE Dark Theme color issues (windows)

### ISSUE THREAD

**Issue by jukzi** [eclipse-platform/.github#45]
"Dark" Theme looks different (wrong) then "Light" Theme . Screenshots from windows 10:
![image](https://user-images.githubusercontent.com/51790620/182793365-cf610327-62d8-4775-a5b5-a9f778ad3281.png)
![image](https://user-images.githubusercontent.com/51790620/182793401-43f1596b-d946-4723-831b-03aba048f37e.png)

1. "Welcome page" is still in light colors
2. Focused Tab (for example "welcome") is blue underlined - while there is no underline in Light theme (https://github.com/eclipse-platform/eclipse.platform.swt/issues/55)
3.the opposite is true for "Run Configurations" Dialog: in dark theme focused Tab "Main" is NOT blue, while in Light them it IS overlined
![image](https://user-images.githubusercontent.com/51790620/182795085-39cae3f2-b496-4dba-b1fb-45777c0c4c68.png)
![image](https://user-images.githubusercontent.com/51790620/182795150-515aac95-3805-41a7-9620-023d7ea51c4c.png)
4. the text input fields (for example "Location" in "Run Configurations") have wrong background color. They are not as black in Dark mode as white in Light
5. same is true for combo boxes (like "Run a product") where one can see the background color that should be used behind the drop down arrow
6. Also sometimes(!) the combos are sometimes blue in dark mode (for example "Launch with" when "Plug-ins" are selected) while not blue in light theme in the same situation:
![image](https://user-images.githubusercontent.com/51790620/182796611-76972faf-8567-4495-9b65-d57e5666b763.png)
![image](https://user-images.githubusercontent.com/51790620/182796677-74b7ea4b-e9c5-4aab-81b9-75f0930f3f55.png)
7. The Tracing Options are hardly readable (grey on grey):
![image](https://user-images.githubusercontent.com/51790620/182797501-65c1f1fd-1e1a-41b9-9c50-2c2e0a90e7f3.png)
8. deactivated combo boxes are hardly readable too:
![image](https://user-images.githubusercontent.com/51790620/182797901-27f7f386-fb0e-43d9-8049-a8025cb51226.png)

**Comment by vogella** [comment]
The CSS settings are located in org.eclipse.ui.themes in the eclipse.platform.ui repo and we rely on SWT API to behave dark aware. The welcome screen is dark for me, see below, please create a new issue for eclipse.platform.ua and ask @BeckerWdf to look at it, IIRC he implemented the dark support for the welcome screen. Might be that Win10 broke something.

The blue highlight of the tab can be configured, @mickaelistria implemented it in SWT. I think we "only" need to active it or turn the CSS specific setting of in the dark theme to activate it.

**Comment by vogella** [comment]
![image](https://user-images.githubusercontent.com/139910/182804900-53469c66-bac7-4a4c-a4c3-5a465f609170.png)

**Comment by jukzi** [comment]
> The CSS settings are located in org.eclipse.ui.themes in the eclipse.platform.ui repo

In dark theme a `new Composite(parent, style)` shows another backgroundcolor then anything that _extends_ Composite. Even a anonymous `new Composite(parent, style) { /** nothing changed **/}` already shows wrong background.
And there are many extensions of Composite in eclipse.... how does that work? Why does any extension of Composite behaves different? Are the classnames  listed in the  e4-dark_globalstyle.css textual compared somewhere instead of `instanceof`?  should be `java.lang.Class.isAssignableFrom(Class<?>)`

**Comment by RedeemerSK** [comment]
Another issue, regression from 2022-06 (4.24) to 2022-09 (4.25):
Background color of buttons/icons does not change neither on mouse hover nor when being enabled (for toggles) / pressed (on action buttons):

![eclipse_2022-06-dark](https://user-images.githubusercontent.com/36334098/197290914-bf21184e-f7cf-4b8a-a381-5f762522305c.png)(2022-06)

![eclipse_2022-09-dark](https://user-images.githubusercontent.com/36334098/197290933-b5c566ab-a9e5-4eba-815f-4d2618d615fd.png)(2022-09)

It seems backgrounds of toggle & action buttons on every view are broken. Not only on hover / press, but also default background is lighter then rest of local toolbar background (see 2nd picture, 3 buttons in "Package Explorer" toolbar).

On main toolbar however, some buttons are ok (eg. "Search" also with dropdown menu arrow or "Skip All Breakpoints" which works in main toolbar even if it's broken in "Breakpoints" view), some are half-ok (eg. for "New" only the dropdown menu arrow background seems ok, for icon itself it's broken), and some are broken (eg. perspectives icons or "Open Terminal" icon).

Minime & Maximize & Restore buttons of views are all working (as well as views icons when minimized).

In short, it's quite inconsistent where the issue manifests, however some patterns are there.

(Windows 10)

**Comment by akurtakov** [comment]
Closing due to no response.
